src/ledger-bridge-keyring.test.ts

```diff
@@ -0,0 +1,845 @@
+import { strict as assert } from 'assert';
+import chai from 'chai';
+import spies from 'chai-spies';
+import EthereumTx from 'ethereumjs-tx';
+import HDKey from 'hdkey';
+import * as ethUtil from 'ethereumjs-util';
+import { TransactionFactory } from '@ethereumjs/tx';
+import Common from '@ethereumjs/common';
+import sigUtil from 'eth-sig-util';
+import windowShim from '../test/window.shim';
+import documentShim from '../test/document.shim';
+
+import { AccountDetails, LedgerBridgeKeyring } from './ledger-bridge-keyring';
+
+global.document = documentShim;
+global.window = windowShim;
+
+const { expect } = chai;
+
+const fakeAccounts = [
+  '0xF30952A1c534CDE7bC471380065726fa8686dfB3',
+  '0x44fe3Cf56CaF651C4bD34Ae6dbcffa34e9e3b84B',
+  '0x8Ee3374Fa705C1F939715871faf91d4348D5b906',
+  '0xEF69e24dE9CdEe93C4736FE29791E45d5D4CFd6A',
+  '0xC668a5116A045e9162902795021907Cb15aa2620',
+  '0xbF519F7a6D8E72266825D770C60dbac55a3baeb9',
+  '0x0258632Fe2F91011e06375eB0E6f8673C0463204',
+  '0x4fC1700C0C61980aef0Fb9bDBA67D8a25B5d4335',
+  '0xeEC5D417152aE295c047FB0B0eBd7c7090dDedEb',
+  '0xd3f978B9eEEdB68A38CF252B3779afbeb3623fDf',
+  '0xd819fE2beD53f44825F66873a159B687736d3092',
+  '0xE761dA62f053ad9eE221d325657535991Ab659bD',
+  '0xd4F1686961642340a80334b5171d85Bbd390c691',
+  '0x6772C4B1E841b295960Bb4662dceD9bb71726357',
+  '0x41bEAD6585eCA6c79B553Ca136f0DFA78A006899',
+  '0xf37559520757223264ee707d4e3fdfaa118db9bd',
+] as const;
```

Comment:
Nit: Do we need this array to be an exact length, and/or is a function expecting an argument with a type of, say, `'0xf37559520757223264ee707d4e3fdfaa118db9bd'`? If not, it seems like this wouldn't need to be `as const`.

Reply:
We can remove `as const` but then the type of `fakeAccounts` would be `Array<string | undefined>`, so we'll have to use an optional chain or a cast on each test where we use those, even if we know they are valid